### PR TITLE
Fix openai not conditioned imports

### PR DESCRIPTION
**Pull request by dakinggg**
Also adds a smoketest to prevent this issue in the future

Also fixes a typo in the code itself

### CODE REVIEW

File: .github/workflows/smoketest.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  run:
+    working-directory: .
+jobs:
+  smoketest:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        python_version:
+        - "3.9"
```
Comment (j316chuck) [inline .github/workflows/smoketest.yaml:26]: Do we always support 3.9?

File: tests/test_smoketest.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+from llmfoundry import callbacks, data, models, optim, tokenizers, utils
+
+
+# This very simple test is just to use the above imports, which check and make sure we can import all the top-level
+# modules from foundry. This is mainly useful for checking that we have correctly conditionally imported all optional
+# dependencies.
+def test_smoketest():
```
Comment (j316chuck) [inline tests/test_smoketest.py:10]: Would put comments as part of function docstring

**Review by j316chuck — APPROVED** [review]
LGTM! Left a question and a nit

File: .github/workflows/smoketest.yaml (hunk 1 above)
Comment (dakinggg) [reply .github/workflows/smoketest.yaml:26]: we should, yeah.

Comment (dakinggg) [reply .github/workflows/smoketest.yaml:26]: we run code quality tests on 3.9 and 3.10 as well

**Review by bandish-shah — APPROVED** [review]
LGTM

**Pull request merged**